src/securejoin.rs

```diff
@@ -616,12 +616,17 @@ pub(crate) async fn observe_securejoin_on_other_device(
                 peerstate.prefer_encrypt = EncryptPreference::Mutual;
                 peerstate.save_to_db(&context.sql).await.unwrap_or_default();
 
-                ChatId::set_protection_for_contact(
-                    context,
-                    contact_id,
-                    mime_message.timestamp_sent,
-                )
-                .await?;
+                if context
+                    .get_config_bool(Config::VerifiedOneOnOneChats)
+                    .await?
+                {
+                    ChatId::set_protection_for_contact(
+                        context,
+                        contact_id,
+                        mime_message.timestamp_sent,
+                    )
+                    .await?;
+                }
```

Comment:
The protected status of chats is still maintained even if `Config::VerifiedOneOnOneChats` is not set, so it was correct not to check for `Config::VerifiedOneOnOneChats`. `Config::VerifiedOneOnOneChats` only means that the UI has support for it (in hindsight, this wasn't a good name for this config). Also see

https://github.com/deltachat/deltachat-core-rust/blob/582541634f806c3044d41cc54e6b0d0e6f8cb081/src/receive_imf.rs#L801-L824

Also, this doesn't have anything to do with sort timestamps, and in general, it's better to keep the PRs as small as possible in order to make reviews faster. It's a bit unfortunate that I have so limited time right now, which makes reviews take a long time, but you can help by splitting up your PRs as much as possible - it's way easier to reason about changes if I can reason about them one at a time.

Reply:
Ok, let's not discuss this here. I added this commit because it textually depends on the previous ones. Probably the right way in such cases is to create another one PR from one of my branches to another (to avoid self-conflicts). I will remove it, but let's leave `refactor: Move calc_{protection_msg_,}sort_timestamp() to 'impl ChatId'` here, it's just a minor refactoring

Reply:
Moved to https://github.com/deltachat/deltachat-core-rust/pull/5107